motor-control-module/src/main/java/org/sagebionetworks/research/motor_control_module/show_step_fragment/ShowTappingCompletionStepFragment.java

```diff
@@ -0,0 +1,146 @@
+package org.sagebionetworks.research.motor_control_module.show_step_fragment;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.StringRes;
+import android.view.View;
+
+import android.widget.TextView;
+import org.sagebionetworks.research.domain.result.interfaces.TaskResult;
+import org.sagebionetworks.research.domain.step.interfaces.SectionStep;
+import org.sagebionetworks.research.domain.step.interfaces.Step;
+import org.sagebionetworks.research.mobile_ui.show_step.view.ShowStepFragmentBase;
+import org.sagebionetworks.research.mobile_ui.show_step.view.ShowUIStepFragmentBase;
+import org.sagebionetworks.research.motor_control_module.R;
+import org.sagebionetworks.research.motor_control_module.result.TappingResult;
+import org.sagebionetworks.research.motor_control_module.show_step_fragment.tapping.TappingButtonIdentifier;
+import org.sagebionetworks.research.motor_control_module.show_step_fragment.tapping.TappingSample;
+import org.sagebionetworks.research.motor_control_module.step.HandStepHelper;
+import org.sagebionetworks.research.motor_control_module.step.TappingStep;
+import org.sagebionetworks.research.motor_control_module.step_view.TappingCompletionStepView;
+import org.sagebionetworks.research.motor_control_module.widget.TapCountResultView;
+import org.sagebionetworks.research.presentation.model.interfaces.StepView;
+import org.sagebionetworks.research.presentation.show_step.show_step_view_models.ShowUIStepViewModel;
+
+import java.util.List;
+
+public class ShowTappingCompletionStepFragment extends
+        ShowUIStepFragmentBase<TappingCompletionStepView, ShowUIStepViewModel<TappingCompletionStepView>,
+                TappingCompletionStepViewBinding> {
+    public static ShowTappingCompletionStepFragment newInstance(@NonNull StepView stepView) {
+        ShowTappingCompletionStepFragment fragment = new ShowTappingCompletionStepFragment();
+        Bundle args = ShowStepFragmentBase.createArguments(stepView);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    protected int getLayoutId() {
+        return R.layout.mpower2_tapping_completion_step;
+    }
+
+    @NonNull
+    @Override
+    protected TappingCompletionStepViewBinding instantiateAndBindBinding(View view) {
+        return new TappingCompletionStepViewBinding(view);
+    }
+
+    @Override
+    public void update(TappingCompletionStepView stepView) {
+        super.update(stepView);
+        TapCountResultView leftResultView = this.stepViewBinding.getLeftResult();
```

Comment:
The code for finding task results and providing TapCountResultView should be in the ViewModel.